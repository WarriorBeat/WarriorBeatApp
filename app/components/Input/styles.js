/**
 * Input/styles.js
 * Styles for Input Components
 * components
 */

import { StyleSheet } from "react-native"
import { colors } from "config/styles"
import {
  heightPercentageToDP as hp,
  widthPercentageToDP as wp,
} from "react-native-responsive-screen"

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 0,
    paddingVertical: hp("3%"),
    borderWidth: 0,
  },
  inputContainer: {
    marginTop: hp("1%"),
    backgroundColor: colors.white,
    borderBottomWidth: 0,
    height: hp("7%"),
    shadowColor: colors.primary,
    shadowOpacity: 0.25,
    shadowRadius: 10,
    shadowOffset: { width: 0, height: 0 },
    elevation: 8,
    borderRadius: wp("4%") / 4,
  },
  input: {
    paddingHorizontal: wp("3%"),
  },
})

export default styles
