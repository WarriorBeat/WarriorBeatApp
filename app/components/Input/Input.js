/**
 * Input.js
 * Text Input Component
 * components
 */

import React from "react"
import { Input as REInput } from "react-native-elements"
import Text from "components/Text"
import styles from "./styles"

const Input = (props) => {
  const {
    label, hideText, name, handleChange,
  } = props
  const textLabel = (
    <Text Color="black_light" Type="header">
      {label}
    </Text>
  )
  return (
    <REInput
      containerStyle={styles.container}
      inputContainerStyle={styles.inputContainer}
      inputStyle={styles.input}
      underlineColorAndroid="transparent"
      {...props}
      label={textLabel}
      clearButtonMode="always"
      secureTextEntry={!!hideText}
      onChangeText={val => handleChange(name, val)}
    />
  )
}

export default Input
