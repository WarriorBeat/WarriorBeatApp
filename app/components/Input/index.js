/**
 * Input/index.js
 * Index Exports for Input Component
 * components
 */

import Input from "./Input"

export default Input
