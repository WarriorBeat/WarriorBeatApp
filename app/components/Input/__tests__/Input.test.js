/**
 * __tests__/Input.test.js
 * Component Test
 * tests
 */

import React from "react"
import { shallow } from "enzyme"

import Input from "components/Input"

describe("Testing Input Component", () => {
  it("renders as expected", () => {
    const wrapper = shallow(<Input label="Label" />)
    expect(wrapper).toMatchSnapshot()
  })
})
