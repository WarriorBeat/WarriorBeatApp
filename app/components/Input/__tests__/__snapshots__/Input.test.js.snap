// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Input Component renders as expected 1`] = `
<ForwardRef(Themed.Input)
  clearButtonMode="always"
  containerStyle={
    Object {
      "borderWidth": 0,
      "paddingHorizontal": 0,
      "paddingVertical": 40,
    }
  }
  inputContainerStyle={
    Object {
      "backgroundColor": "rgba(255,255,255,1)",
      "borderBottomWidth": 0,
      "borderRadius": 7.5,
      "elevation": 8,
      "height": 93.5,
      "marginTop": 13.5,
      "shadowColor": "#393939",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 10,
    }
  }
  inputStyle={
    Object {
      "paddingHorizontal": 22.5,
    }
  }
  label={
    <Text
      Color="black_light"
      Italic={false}
      NoPadding={false}
      Shadow={false}
      Type="header"
      Uppercase={false}
      Weight={false}
      fontSize={null}
      style={Object {}}
    >
      Label
    </Text>
  }
  onChangeText={[Function]}
  secureTextEntry={false}
  underlineColorAndroid="transparent"
/>
`;
